File: app/src/main/java/com/wing/baseframe/utils/BarUtils.java
package com.wing.baseframe.utils;

import android.app.Activity;
import android.view.View;

public final class BarUtils {

    /**
     * 隐藏状态栏和虚拟按键拦
     *
     * @param activity
     */
    public static void hideSystemUI(Activity activity) {
        if (activity == null) return;
        final View decorView = activity.getWindow().getDecorView();
        final int flags = View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
                | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
                | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
                | View.SYSTEM_UI_FLAG_FULLSCREEN
                | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY;
        decorView.setSystemUiVisibility(flags);
        decorView.setOnFocusChangeListener(new View.OnFocusChangeListener() {
            @Override
            public void onFocusChange(View v, boolean hasFocus) {
                decorView.setSystemUiVisibility(flags);
            }
        });
    }

}

File: app/src/main/java/com/wing/baseframe/utils/BarUtils.kt
package com.wing.baseframe.utils

import android.app.Activity
import android.view.View

/**
 *姓名：mengc
 *日期：2019/1/8
 *功能：
 */

object BarUtils {
    /**
     * 隐藏状态栏和虚拟按键拦
     *
     * @param activity
     */
    fun hideSystemUI(activity: Activity?) {
        if (activity == null) return
        val decorView = activity.window.decorView
        val flags = (View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                or View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
                or View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
                or View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
                or View.SYSTEM_UI_FLAG_FULLSCREEN
                or View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY)
        decorView.systemUiVisibility = flags
        decorView.onFocusChangeListener =
                View.OnFocusChangeListener { v, hasFocus -> decorView.systemUiVisibility = flags }
    }
}